modules/core/src-2/NewtypeValidated.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+abstract class NewtypeValidated[A] extends HasId { self =>
+  // This encoding originally comes from this library:
+  // https://github.com/alexknvl/newtypes#what-does-it-do
+  type Base
+  trait _Tag extends Any
+  type Type <: Base with _Tag
+
+  @inline def apply(a: A): Either[String, Type]
+
+  @inline final def unsafeApply(a: A): Type = a.asInstanceOf[Type]
+
+  @inline final def value(x: Type): A =
+    x.asInstanceOf[A]
+
+  implicit final class Ops(val self: Type) {
+    @inline final def value: A = NewtypeValidated.this.value(self)
+  }
+
+  def schema: Schema[Type]
+
+  implicit val tag: ShapeTag[Type] = new ShapeTag[Type] {
+    def id: ShapeId = self.id
+    def schema: Schema[Type] = self.schema
+  }
+
+  def unapply(t: Type): Some[A] = Some(t.value)
+
+  implicit def refinementProvider[C](implicit
+      ev: RefinementProvider.Simple[C, A]
+  ): RefinementProvider.Simple[C, Type] =
+    new RefinementProvider.Simple[C, Type] {
+      val tag: ShapeTag[C] = ev.tag
+      override def make(c: C): Refinement.Aux[C, Type, Type] =
+        ev.make(c).imapFull[Type, Type](asBijectionUnsafe, asBijectionUnsafe)
+    }
+
+  protected val validators: List[A => Either[String, A]]
+
+  protected def validateInternal[C](c: C)(value: A)(implicit
+      ev: RefinementProvider.Simple[C, A]
+  ): Either[String, A] =
+    ev.make(c).apply(value)
```

Comment:
I'm not liking this. For starters, the `validateInternal` re-instantiates the refinement on a per-element basis. Also, the fact that a validation may be composed of several validators is an implementation detail that should not appear in the interface of `NewTypeValidated`

You could create a `smithy4s.Validator` interface that `Refinement` would extend, and enrich it with a composition method (andThen). 

The renderer would be responsible for rendering something that could look like 

```scama
val validator : Validator[A] = Validator.of[A].validating(C1).and(C2).and(C3).build()
def apply(a: A) : Either[String, Type] = {
   validator(a) 
}
```

Reply:
It's pretty close. See my comments on the file in question.